pkg/binding/add.go

```diff
@@ -81,3 +105,90 @@ func (o *BindingClient) AddBinding(bindingName string, bindAsFiles bool, unstruc
 
 	return libdevfile.AddKubernetesComponentToDevfile(string(yamlDesc), serviceBinding.Name, obj)
 }
+
+func (o *BindingClient) AddBinding(
+	flags map[string]string,
+	bindingName string,
+	bindAsFiles bool,
+	unstructuredService unstructured.Unstructured,
+	workloadName string,
+	workloadGVK schema.GroupVersionKind,
+) error {
+	service, err := o.kubernetesClient.NewServiceBindingServiceObject(unstructuredService, bindingName)
+	if err != nil {
+		return err
+	}
+
+	serviceBinding := kclient.NewServiceBindingObject(bindingName, bindAsFiles, workloadName, workloadGVK, []sboApi.Mapping{}, []sboApi.Service{service}, sboApi.ServiceBindingStatus{})
+
+	var backend backendpkg.AddBindingBackend
+	if len(flags) == 0 {
+		backend = o.interactiveBackend
+	} else {
+		backend = o.flagsBackend
+	}
+
+	var options []asker.CreationOption
+	for len(options) == 0 {
+		options, err = backend.SelectCreationOptions(flags)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Note: we cannot directly marshal the serviceBinding object to yaml because it doesn't do that in the correct k8s manifest format
+	serviceBindingUnstructured, err := kclient.ConvertK8sResourceToUnstructured(serviceBinding)
```

Comment:
Have you explored `runtime.DefaultUnstructuredConverter.ToUnstructured(serviceBinding)`? I used its counterpart in `odo logs` https://github.com/redhat-developer/odo/blob/b3517f285e9e1a3a19996401f7e7d5e494986d2e/pkg/logs/logs.go#L76

Reply:
There is  a TODO in `ConvertK8sResourceToUnstructured` to use this method instead

I prefer if it is done as part of another PR, to do this in a consistent manner across all the code

Reply:
Sure, makes sense. I didn't notice the TODO.